File: solutions/java/bird-watcher/1/src/main/java/BirdWatcher.java

class BirdWatcher {
    private final int[] birdsPerDay;

    public BirdWatcher(int[] birdsPerDay) {
        this.birdsPerDay = birdsPerDay.clone();
    }

    public int[] getLastWeek() {
        return new int[] {0, 2, 5, 3, 7, 8, 4};
    }

    public int getToday() {
        int i = birdsPerDay.length-1;
        return birdsPerDay[i];
    }

    public void incrementTodaysCount() {
        int i = birdsPerDay.length-1;
        birdsPerDay[i]++;
    }

    public boolean hasDayWithoutBirds() {
        
        for(int i = 0; i <= birdsPerDay.length-1; i++){
            
            if(birdsPerDay[i] == 0){
                return true;
            } 
        } return false;
    }

    public int getCountForFirstDays(int numberOfDays) {
        int sum = 0;
        int daysToCount = Math.min(numberOfDays, birdsPerDay.length);

        for(int i = 0; i<daysToCount; i++){
            sum += birdsPerDay[i];
        };
        return sum;
    }

    public int getBusyDays() {
        int busyDays = 0;
        for(int i = 0; i < birdsPerDay.length; i++){
            if(birdsPerDay[i] >= 5){
                busyDays++;
            }
        } return busyDays;
    }
}

File: solutions/csharp/bird-watcher/1/BirdWatcher.cs
class BirdCount
{
    private int[] birdsPerDay;

    public BirdCount(int[] birdsPerDay)
    {
        this.birdsPerDay = (int[])birdsPerDay.Clone();
    }

    public static int[] LastWeek() => [0, 2, 5, 3, 7, 8, 4];

    public int Today() => birdsPerDay[birdsPerDay.Length-1];

    public void IncrementTodaysCount() => birdsPerDay[birdsPerDay.Length-1]++;

    public bool HasDayWithoutBirds() => Array.IndexOf(birdsPerDay, 0) != -1;

    public int CountForFirstDays(int numberOfDays){
        int sum=0;
        for(int i = 0; i < numberOfDays; i++){
           sum+=birdsPerDay[i]; 
        }
        return sum;
    }

    public int BusyDays()
    {
        List<int> days = new List<int>(birdsPerDay);
        List<int> count = days.FindAll(number => number >= 5);
        return count.Count();
    }
}
